docs/source/design/ui_design.rst

```diff
@@ -255,69 +255,85 @@ dictionary with a schema defined by QCElemental.
+keyword arguments or by using the ``options`` argument, which is an object that
+holds all the possible options with reasonable default values. Users can
+grab this object and figure out the available options and their default values
+and modify them as they wish.
+
+With the ``calculate()`` function, a user can run the H2 scf/sto-3g example by
+specifying only the required arguments as shown below.
 
 .. code-block:: python
 
-    def calculate_scf_energy(molecule: Union[str, chemist.Molecule], basis:
-    Union[str, chemist.AOBasisSet, simde.type.ao_space], spin: int = 1, max_iterations: int = 50,
-    scf_thresh: float = 1e-10, options: Union[Dict, dataclass] = None, **kwargs) -> float:
-
-In this function, the only required arguments are ``molecule``, which can either
-be a Python ``string`` or a ``chemist.Molecule`` object and ``basis``, which can
-either be a a Python ``string`` or a ``chemist.AOBasisSet`` or a
-``simde.type.ao_space``. Additional SCF related options are also explicitly
-defined in the function with reasonable default values. While the number of
-options are limited for SCF, for correlated methods more options may need to be
-defined on top of the SCF related options. One solution to avoid a long list of
-arguments is to provide additional options through ``**kwargs`` argument at the
-end. However, this would require users to know the correct keywords. To help our
-users to figure out the available keywords, we also provide the ``options``
-argument, which is an object that holds all the possible options with reasonable
-default values. Our users can grab this object and figure out the available
-options and their default values and modify them as they wish. 
-
-Similarly, ``calculate_X_energy`` functions will also be provided for all other
-quantum chemistry methods supported by NWChemEx. Users may also request other
-properties such as the gradients and the wave function using any of the methods. 
-These can be obtained by ``calculate_X_gradients`` and
-``calculate_X_wavefunction``. It should be noted that, caching mechanism of
-NWChemEx avoids any recomputation if the requested quantity is already calculated.
-
-Our UI also provides a more general ``nwx.calculate`` function, where the method
-and the return types can be specified in the ``options`` explicitly or provided
-as key-value arguments in the function call.
+    import nwchemex as nwx 
+    energy = nwx.calculate(molecule = 'H 0. 0. 0. \n H 0., 0. 1.', method='scf', basis = 'sto-3g')
+
+While this function can be enough for most common tasks, the flexibility of this
+function is limited particularly for the additional outputs that a user might be
+interested in, such as the wavefunction, density matrix, or intermediate outputs
+such as the data related to the convergence or timings of the SCF iterations. To
+address this issue, we provide a more flexible ``run()`` function with the
+following signature.
 
 .. code-block:: python
 
```

Comment:
```suggestion
```
IMHO having two similarly named functions, `compute` and `run`, which do similar things is confusing. I'm also not sure I understand the distinction between `nwx.Simulation` and the opaque `Options` object (other than `nwx.Simulation` seems to also hold the molecule, basis set). The molecule and basis set are going to need to be unpacked to call PluginPlay, so packing them into a new object just to unpack them seems heavy.

Reply:
> IMHO having two similarly named functions, compute and run, which do similar things is confusing.

I agree about having both to be confusing. I wanted to have a discussion about this alternative. We can select one of them at the end or come up with something different.

My problem with the current `calculate()` function is the return type. Having return types depending on the options is not ideal especially if we want to enable returning other properties as well. So, the main distinction of `Simulation` is to hold both input (`options`) and output. So, the users have an object at the end of a calculation to save, unpack as they wish.

> The molecule and basis set are going to need to be unpacked to call PluginPlay, so packing them into a new object just to unpack them seems heavy.

I am not sure if it adds any significant overhead, but `run` function could also take `molecule`, `basis set` etc as required options to avoid unpacking cost. They can be packed into the `Simulation` object during the function call if the user wants to save the results. So, there will be only packing cost if needed, no unpacking.

Reply:
> My problem with the current calculate() function is the return type. Having return types depending on the options is not ideal especially if we want to enable returning other properties as well. So, the main distinction of Simulation is to hold both input (options) and output. So, the users have an object at the end of a calculation to save, unpack as they wish.

Gotcha. Those are valid concerns. That said, I generally recommend designing pure functions. Part of being a pure function is treating the inputs to the function exclusively as inputs (i.e., not using them as returns too). So I still think the function should return its returns. At a minimum I'd recommend an API like:

```.py
results = nwx.calculate(inputs)
```

Where both `inputs` and `results` are dictionary-like. This is actually very similar to the API of `Module::run` (not `run_as`; for `run_as`  the user passes unpacked inputs, and receives unpacked returns). The key difference between the proposed `nwx.calculate` and  `Module::run` is that `Module::run` only concerns itself with inputs/results directly to that module (not the submodules). The proposed `ModuleGraph` object in [PluginPlay](https://github.com/NWChemEx-Project/PluginPlay/issues/308) would allow setting inputs and getting results from an entire graph. Hence `nwx.calculate` would just be a thin wrapper around `ModuleGraph::run` when it exists. 

I admittedly had been fixated on calling `run_as` which is why I thought we would have to unpack the inputs to call PluginPlay. If we instead target `Module::run` (and presumably `ModuleGraph::run` when it exists) then my previous statement is wrong (*i.e.*, we don't need to unpack the inputs to call PluginPlay). FWIW, on the C++-side we usually don't use `Module::run` directly on account of it needing to take type-erased inputs, which is cumbersome to prepare. However, Python objects are already type-erased. So it should be straightforward to use `Module::run` directly from Python.

That all said, to be user-friendly, I still think `nwx.calculate` will need to treat the molecule, and basis set specially, and thus to distinguish that fact I'd advocate for "molecule" and "basis" being separate arguments to `nwx.calculate`. More specifically, we'll want to support some string-based representations of the molecule and basis set. If we insist that the method specification maps to a key known to the `ModuleManager` (I strongly favor this option), then we don't actually need to do anything special for it. However, if we don't want to enforce such a consideration, then "method" should be separate too. My guess is that if "molecule" and "basis" are separate, users will probably expect "method" to show up explicitly too. So I think that gets us back to an API like:

```.py
results = nwx.calculate(mol = '0 1\nH 0.0 0.0 0.0\nH 0.0 0.0 0.89', method = 'scf', basis = 'aug-cc-pVDZ', additional_inputs)
```

where `additional_inputs` and `results` are (at this design point) opaque types. I'm okay having both `additional_inputs` and `results` be dictionary-like, even if there is a single input or result; for `results` I'm also okay automatically unpacking a single result (the user presumably knows what they asked for, so they should know what it is).